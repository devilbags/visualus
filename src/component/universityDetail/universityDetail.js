// @flow
import React from 'react';
// import type { University } from '../../type/university'

import './universityDetail.css';

// type Props = {
//     university: University
// }

const UniversityDetail = () => {
  return <h1>Display University detail here</h1>;
};

export default UniversityDetail;
