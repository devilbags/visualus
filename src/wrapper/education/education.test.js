import React from 'react';
import { shallow, mount } from 'enzyme';
import { StaticRouter, MemoryRouter, BrowserRouter } from 'react-router-dom';
import Education from './education';
import University from '../../component/university/university';
import UniversityDetail from '../../component/universityDetail/universityDetail';
import universityDetail from '../../component/universityDetail/universityDetail';

describe('Education wrapper', () => {
  const defaultProps = {
    loading: false,
    universityList: [],
    onInputValueChange: jest.fn(),
    onUniversitySelected: jest.fn()
  };
  let wrapper;
  it('should render universityList component', () => {
    wrapper = mount(
      <MemoryRouter initialEntries={['/education']}>
        <Education
          loading={defaultProps.loading}
          universityList={defaultProps.universityList}
          onInputValueChange={defaultProps.onInputValueChange}
        />
      </MemoryRouter>
    );
    expect(wrapper.find(University)).toHaveLength(1);
    expect(wrapper.find(universityDetail)).toHaveLength(0);
  });
  it('should not render universitydetail component when there is no selected', () => {
    wrapper = mount(
      <MemoryRouter initialEntries={['/education/detail']}>
        <Education
          loading={defaultProps.loading}
          universityList={defaultProps.universityList}
          onInputValueChange={defaultProps.onInputValueChange}
        />
      </MemoryRouter>
    );
    expect(wrapper.find(University)).toHaveLength(0);
    expect(wrapper.find(UniversityDetail)).toHaveLength(0);
  });
  it('should not render universitydetail component when selected', () => {
    const selectedU = {};
    wrapper = mount(
      <MemoryRouter initialEntries={['/education/detail']}>
        <Education
          loading={defaultProps.loading}
          universityList={defaultProps.universityList}
          onInputValueChange={defaultProps.onInputValueChange}
          selectedUniversity={selectedU}
        />
      </MemoryRouter>
    );
    expect(wrapper.find(University)).toHaveLength(0);
    expect(wrapper.find(UniversityDetail)).toHaveLength(1);
  });
});
