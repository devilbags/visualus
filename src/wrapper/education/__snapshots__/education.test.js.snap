// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education wrapper Should match shallow snapshot 1`] = `
<Adapter
  style={
    Object {
      "padding": "8px",
    }
  }
>
  <div
    className="education"
  >
    <University />
  </div>
</Adapter>
`;
